src/normalizer/mod.rs

```diff
@@ -57,13 +65,35 @@ impl<'o> Iterator for NormalizedTokenIter<'o> {
     }
 }
 
+/// Structure for providing options to a normalizer.
+#[derive(Clone)]
+pub struct NormalizerOption {
+    pub create_char_map: bool,
+}
+
+impl NormalizerOption {
+    
+    /// Creates a new [`NormalizerOption`] with default options set.
+    pub fn new() -> NormalizerOption {
+        NormalizerOption {
+            create_char_map: true,
+        }
+    }
+}
+
 /// Trait defining a normalizer.
 pub trait Normalizer: Sync + Send {
     /// Normalize the provided [`Token`].
     ///
     /// A Normalizer can return several `Token`s.
     fn normalize<'o>(&self, token: Token<'o>) -> Box<dyn Iterator<Item = Token<'o>> + 'o>;
 
+    /// Normalize the provided [`Token`].
+    /// Options can be set using the provided [`NormalizerOption`].
+    ///
+    /// A Normalizer can return several `Token`s.
+    fn normalize_with_option<'o>(&self, token: Token<'o>, options: NormalizerOption) -> Box<dyn Iterator<Item = Token<'o>> + 'o>;
```

Comment:
You can provide a default implementation directly in the trait, that would avoid reimplementing it when it is not necessary.
```suggestion
    fn normalize_with_option<'o>(&self, token: Token<'o>, options: NormalizerOption) -> Box<dyn Iterator<Item = Token<'o>> + 'o> {
        self.normalize(token)
    }
```

Reply:
However, we could always pass the option to the `normalize` method that will avoid having 2 implementations of the same thing.